cert/spec.go

```diff
@@ -262,34 +271,24 @@ func (spec *Spec) Lifespan() time.Duration {
 		panic("cert: cannot check certificate's lifespan because spec has an invalid transport")
 	}
 
-	// This bit of code is necessary to confirm that the cert/key are older than the spec definition.
-	if spec.IsChangedOnDisk(spec.Key.Path) || spec.IsChangedOnDisk(spec.Cert.Path) {
-		// This is necessary to essentially force cfssl to regenerate since it's not spec aware.
-		log.Infof("refreshing due to spec %s having a newer mtime than key or cert", spec.Path)
-		spec.ForceRenewal()
-		return 0
-	}
 	return spec.tr.Lifespan()
 }
 
-// IsChangedOnDisk method to report if the given path is older than the spec
-func (spec *Spec) IsChangedOnDisk(path string) bool {
+// HasChangedOnDisk returns (removed, changed, err) to indicate if the spec has changed
+func (spec *Spec) HasChangedOnDisk() (bool, bool, error) {
 	specStat, err := os.Stat(spec.Path)
-	if err != nil {
-		// The assertion here is that the spec actually
-		// exists. If it doesn't, something is wrong with the
-		// world.
-		log.Warning("cert: IsChangedOnDisk: Spec file does not exist")
-		return true
-	}
-	st, err := os.Stat(path)
 	if err != nil {
 		if os.IsNotExist(err) {
-			log.Errorf("cert isChangedOnDisk: while checking path %s, got path error %s", path, err)
+			log.Debugf("spec %s was removed from on disk", spec)
+			return true, false, nil
 		}
-		return true
+		return false, false, err
+	} else if specStat.ModTime().After(spec.loadTime) {
```

Comment:
nit: The `else` is unnecessary here, and conventional Go style for checking `err`  is to drop the else, add vertical space, and then put the next branch.

Reply:
go linter don't care about that, so I'm undecided if I should care (if it was a raw 'if else', meh- I get their reasons but it's dogma in my view, but this isn't quite that).  Either way....

I'll be back in this pathway soon enough- the logic here assumes that mtime means "it's changed"; that doesn't, the content may be the same.  I was intending on adding parse comparisons to properly handle this in a follow up PR.  Will revisit it then.

Reply:
Yeah, I don't have any objective reason other than it's convention. 

```
if err != nil {
    //...
    return
}
``` is such commonly-repeated Go boilerplate that I just tend to notice any time it's different.